network/websocket_dataframe.ksy

```diff
@@ -0,0 +1,68 @@
+meta:
+  id: websocket_dataframe
+  title: WebSocket
+  xref:
+    rfc: 6455
+  endian: be
+  license: APACHE-2.0
+doc: |
+  The WebSocket protocol establishes a two-way communication channel via TCP.
+  Messages are made up of one or more dataframes, and are delineated by
+  frames with the `fin` bit set.
+seq:
+  - id: fin
+    type: b1
+  - id: rsv1
+    type: b1
+  - id: rsv2
+    type: b1
+  - id: rsv3
+    type: b1
+  - id: opcode
+    enum: opcode
+    type: b4
+  - id: b1
+    type: u1
+  - id: extended_len_1
+    type: u2
+    if: payload_len == 126
+  - id: extended_len_2
+    type: u4
+    if: payload_len == 127
+  - id: mask_key
+    type: u4
+    if: is_masked == 1
+  - id: payload
```

Comment:
Should be parsed depending on opcode

Reply:
Not sure what you mean here; the masking key is resolved based on an earlier flag, and is unrelated to the opcode.

Reply:
```yaml
- id: reserved
  type: b3
```

Reply:
Ah. Still issues with that; `continuation` opcodes are ambiguous, as the first frame will have determined the message format. Is it worth changing this from a dataframe parser to a message parser?

Reply:
Continuation will be really pain. [`Chains` proposal](https://github.com/kaitai-io/kaitai_struct/issues/196) can solve it, but it is not yet implemented.